util/monoclock.c

```diff
@@ -94,3 +94,35 @@ monoclock_get_cputime(struct timeval * tv)
 	/* Failure! */
 	return (-1);
 }
+
+/**
+ * monoclock_getres(resd):
+ * Store the timer resolution in ${resd}. If CLOCK_MONOTONIC is available, use
```

Comment:
I'd say "Store an upper limit on timer granularity" here -- on many systems gettimeofday has microsecond accuracy.

You should probably have a comment somewhere saying that we're assuming that `clock_getres` works iff `clock_gettime` works.  It's a reasonable assumption to make, but it should still be documented... I could imagine embedded systems without RTC clocks which don't implement `clock_getres(CLOCK_REALTIME)` but nonetheless return a timer granularity for `clock_getres(CLOCK_*)`.